src/cache-utils.ts

```diff
@@ -66,35 +104,123 @@ export const getPackageManagerInfo = async (packageManager: string) => {
+ * traversing through the directories tree
+ */
+export const memoizedCacheDependencies: Record<string, string[]> = {};
+/**
+ * Expands (converts) the string input `cache-dependency-path` to list of directories that
+ * may be project roots
+ * @param cacheDependencyPath - either a single string or multiline string with possible glob patterns
+ *                              expected to be the result of `core.getInput('cache-dependency-path')`
+ * @return list of directories and possible
+ */
+const getProjectDirectoriesFromCacheDependencyPath = async (
+  cacheDependencyPath: string
+): Promise<string[]> => {
+  let cacheDependenciesPaths: string[];
+
+  // memoize unglobbed paths to avoid traversing FS
+  const memoized = memoizedCacheDependencies[cacheDependencyPath];
+  if (memoized) {
+    cacheDependenciesPaths = memoized;
+  } else {
+    const globber = await glob.create(cacheDependencyPath);
+    cacheDependenciesPaths = (await globber.glob()) || [''];
+    memoizedCacheDependencies[cacheDependencyPath] = cacheDependenciesPaths;
+  }
+
+  const existingDirectories: string[] = cacheDependenciesPaths
+    .map(path.dirname)
+    .filter(path => path != null)
+    .filter(unique())
+    .filter(fs.existsSync)
+    .filter(directory => fs.lstatSync(directory).isDirectory());
+
+  // if user explicitly pointed out some file, but it does not exist it is definitely
+  // not he wanted, thus we should throw an error not trying to workaround with unexpected
+  // result to the whole build
+  if (existingDirectories.length === 0)
+    throw Error(
+      'No existing directories found containing `cache-dependency-path`="${cacheDependencyPath}"'
+    );
+
+  return existingDirectories;
+};
+
+/**
+ * Finds the cache directories configured for the repo if cache-dependency-path is not empty
+ * @param packageManagerInfo - an object having getCacheFolderPath method specific to given PM
+ * @param cacheDependencyPath - either a single string or multiline string with possible glob patterns
+ *                              expected to be the result of `core.getInput('cache-dependency-path')`
+ * @return list of files on which the cache depends
+ */
+const getCacheDirectoriesFromCacheDependencyPath = async (
   packageManagerInfo: PackageManagerInfo,
-  packageManager: string
-) => {
-  const stdOut = await getCommandOutput(
-    packageManagerInfo.getCacheFolderCommand
+  cacheDependencyPath: string
+): Promise<string[]> => {
+  const projectDirectories = await getProjectDirectoriesFromCacheDependencyPath(
+    cacheDependencyPath
   );
+  const cacheFoldersPaths = await Promise.all(
+    projectDirectories.map(projectDirectory =>
+      packageManagerInfo
+        .getCacheFolderPath(projectDirectory)
+        .then(cacheFolderPath => {
+          core.debug(
+            `${packageManagerInfo.name}'s cache folder "${cacheFolderPath}" configured for the directory "${projectDirectory}"`
+          );
+          return cacheFolderPath;
+        })
```

Comment:
Could you please change then block to await.

Reply:
done